bash_archiving.sh

```diff
@@ -9,7 +9,24 @@ then
 	then
 		for (( index=1; index <= $countUSERS; index++))
 		do
-			word=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ' | cut -d " " -f $index)
+			word="$(
+				awk -v I=$index '
```

Comment:
This allows awk to set and assign an internal variable the contents of the external variable; put simply, awk can then process dynamic data taken from outside of the awk code.